user/applications/tinker/src/burnin_test.cpp

```diff
@@ -459,5 +473,147 @@ bool BurninTest::testCpuLoad() {
 	return true;
 }
 
+#if PLATFORM_ID == PLATFORM_MSOM
+
+static bool turnModemOn() {
+	Cellular.on();
+	waitFor(Cellular.isOn, 30000);
+
+	if (!Cellular.isOn()) {
+		strcpy(BurninErrorMessage, "Cell modem failed to turn on after 30s");
+		return false;
+	}
+	return true;
+}
+
+bool BurninTest::testCellularModem() {
+	if (!turnModemOn()) {
+		return false;
+	}
+	
+    CellularDevice device = {};
+    device.size = sizeof(device);
+    if (cellular_device_info(&device, NULL)) {
+		strcpy(BurninErrorMessage, "Failed to get cellular info");
+    	return false;
+    }
+    
+    Log.info("Cell modem ICCID: %s IMEI: %s FW: %s", device.iccid, device.imei, device.radiofw);
+
+	Cellular.off();
+	waitFor(Cellular.isOff, 30000);
```

Comment:
Do we want to check if it turns off?  Maybe add `turnModemOff()`?  It could also not explicitly fail the test with `false` if it doesn't turn off.

Reply:
Its not required that the modem be off for the test to be considered successful. This is just an attempt to maintain state after the test is complete. If the modem fails to power on for subsequent cell/gnss tests, that is a legitimate failure though.